src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -125,65 +133,142 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
+
   const handleKeyDown = (event, item) => {
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
       event.preventDefault();
       handleOnClick(event, item);
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_DOWN) {
+      event.preventDefault();
+      handleDownArrow();
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_UP) {
+      event.preventDefault();
+      handleUpArrow();
+    } else if ((event.nativeEvent.keyCode === KeyCode.KEY_RIGHT || event.nativeEvent.keyCode === KeyCode.KEY_LEFT) && !currentNodeId) {
+      event.preventDefault();
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_RIGHT) {
+      event.preventDefault();
+      if (document.getElementById(currentNodeId).ariaExpanded === 'true') {
+        handleDownArrow();
+      } else if (document.getElementById(currentNodeId).ariaHasPopup && (!document.getElementById(currentNodeId).ariaExpanded || document.getElementById(currentNodeId).ariaExpanded === 'false')) {
+        handleOnClick(event, item);
+      }
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_LEFT) {
+      event.preventDefault();
+      if (document.getElementById(currentNodeId).ariaExpanded === 'true') {
+        handleOnClick(event, item);
+      } else if (!document.getElementById(currentNodeId).ariaHasPopup || !document.getElementById(currentNodeId).ariaExpanded || document.getElementById(currentNodeId).ariaExpanded === 'false') {
+        findParent();
+      }
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_HOME) {
+      event.preventDefault();
+      cursor.current = 0;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_END) {
+      event.preventDefault();
+      cursor.current = visibleNodes.current.length - 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    } else if (event.nativeEvent.keyCode >= KeyCode.KEY_A && event.nativeEvent.keyCode <= KeyCode.KEY_Z) {
+      event.preventDefault();
+      const char = String.fromCharCode(event.nativeEvent.keyCode);
+      findNode(char);
+    } else if (event.nativeEvent.shiftKey && event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
+      handleUpArrow();
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
+      handleDownArrow();
     }
   };
 
-  const renderMenuItems = (currentMenuItem, firstLevel) => {
+  const renderMenuItems = (currentMenuItem, parent = '', firstLevel = false) => {
     if (!currentMenuItem) {
       return undefined;
     }
 
     return currentMenuItem.map((item) => {
       const itemIsOpen = openKeys[item.path];
       const itemHasChildren = item.childItems !== undefined;
+      const id = item.name.split(' ').join('-');
       let isSelected = false;
       let selectedRef;
 
       if (selectedPath === item.path) {
         isSelected = true;
         selectedRef = selectedItem;
+      } else if (currentNodeId === id) {
```

Comment:
Should this check be moved into the original if check? Could two items end up selected by accident?

Reply:
The `isSelected` variable is used to decide which nodes should receive emphasis styles (ie the blue highlighting around a menu item), so in this case we actually want more than one node to evaluate to true here. The reason I split these two checks is that we want `selectedRef` to refer to the menu item that is currently open on the page. Making them the same check would cause `selectedRef` to refer to the item the user is currently hovering over through keyboard navigation.

The variable name for `isSelected` is pretty confusing though, I should probably change that 😬.